cmake/Protobuf.cmake

```diff
@@ -1,5 +1,5 @@
 macro(p4c_obtain_protobuf)
-  set(P4C_PROTOBUF_VERSION 25.3)
+  set(P4C_PROTOBUF_VERSION 30.0)
```

Comment:
I guess most older versions of protobuf have problems compiling from source with gcc-15 on Ubuntu 24.04 also?

(Not a complaint from me: I have spent hours trying to find combinations of protobuf and grpc versions that compile from source in the past.)